compiler/src/dotty/tools/dotc/reporting/Message.scala

```diff
@@ -384,6 +383,12 @@ abstract class Message(val errorId: ErrorMessageID)(using Context) { self =>
    */
   def showAlways = false
 
+  /** A list of actions attatched to this message to address the issue this
+    * message represents.
+    */
+  def actions(using Context): java.util.List[CodeAction] =
```

Comment:
I guess we could add it when we need it, but ideally shouldn't our internal APIs let us define a list of WorkspaceEdit for a given message? A concrete example where this would be useful is https://github.com/lampepfl/dotty/pull/16992 where the compiler gives you two possible ways to fix your code.

Reply:
We talked a bit about this in person this morning, but this is totally doable already. The `CodeAction` essentially holds a workspace edit. It needs to be wrapped in the action to ensure that each one has a unique title for the exact situation you're mentioning.